marklogic-data-hub/src/main/java/com/marklogic/hub/impl/DataHubImpl.java

```diff
@@ -895,10 +895,31 @@ protected void prepareProjectBeforeUpgrading(HubProject hubProject, String curre
     // only used in test
     public void setHubConfig(HubConfigImpl hubConfig) {
         this.hubConfig = hubConfig;
+        if (this.loadUserModulesCommand != null) {
```

Comment:
All of these have constructors that take a HubConfig - those should be used instead, particularly for LoadHubModulesCommand, which will then construct its other field using the given HubConfig.